packages/stitch/src/plugins/computedFieldsPlugin.ts

```diff
@@ -0,0 +1,167 @@
+
+import { GraphQLSchema, GraphQLObjectType, GraphQLInterfaceType } from 'graphql';
+
+import { IStitchSchemasOptions, StitchSchemasPlugin } from '../types';
+
+interface ComputedFieldConfig extends MergedFieldConfig {
+  computed?: string;
+}
+
+export class ComputedFieldsPlugin implements StitchSchemasPlugin {
+  public preconfigure(config: IStitchSchemasOptions): IStitchSchemasOptions {
+    if (config.subschemas) {
+      const mapped: Array<GraphQLSchema | SubschemaConfig | Array<SubschemaConfig>> = [];
+
+      config.subschemas.forEach(schemaLikeObject => {
+        if (isSubschemaConfig(schemaLikeObject) && schemaLikeObject.merge) {
+          mapped.push(...isolateComputedMergeSchemas(schemaLikeObject as SubschemaConfig));
+        } else {
+          mapped.push(schemaLikeObject);
+        }
+      });
+
+      config.subschemas = mapped;
+    }
+    return config;
+  }
+}
+
+function isolateComputedMergeSchemas(subschemaConfig: SubschemaConfig): Array<SubschemaConfig> {
+  const staticTypes: Record<string, MergedTypeConfig> = {};
+  const computedTypes: Record<string, MergedTypeConfig> = {};
+
+  Object.keys(subschemaConfig.merge).forEach((typeName: string) => {
+    const mergedTypeConfig: MergedTypeConfig = subschemaConfig.merge[typeName];
+    staticTypes[typeName] = mergedTypeConfig;
+
+    if (mergedTypeConfig.fields) {
+      const staticFields: Record<string, MergedFieldConfig> = {};
+      const computedFields: Record<string, MergedFieldConfig> = {};
+
+      Object.keys(mergedTypeConfig.fields).forEach((fieldName: string) => {
+        const mergedFieldConfig: ComputedFieldConfig = mergedTypeConfig.fields[fieldName] as ComputedFieldConfig;
+
+        if (mergedFieldConfig.selectionSet && mergedFieldConfig.computed) {
+          computedFields[fieldName] = mergedFieldConfig;
+        } else {
+          staticFields[fieldName] = mergedFieldConfig;
+        }
+        delete mergedFieldConfig.computed;
+      });
+
+      const computedFieldCount = Object.keys(computedFields).length;
+      const objectType = subschemaConfig.schema.getType(typeName) as GraphQLObjectType;
+
+      if (computedFieldCount && computedFieldCount !== Object.keys(objectType.getFields()).length) {
+        staticTypes[typeName] = {
+          ...mergedTypeConfig,
+          fields: Object.keys(staticFields).length ? staticFields : undefined,
+        };
+        computedTypes[typeName] = { ...mergedTypeConfig, fields: computedFields };
+      }
+    }
+  });
+
+  if (Object.keys(computedTypes).length) {
+    const endpoint = getSharedEndpoint(subschemaConfig);
+    return [
+      filterComputedSubschema({ ...subschemaConfig, endpoint, merge: computedTypes }),
+      filterStaticSubschema({ ...subschemaConfig, endpoint, merge: staticTypes }, computedTypes),
+    ];
+  }
+
+  return [subschemaConfig];
+}
+
+function getSharedEndpoint(subschemaConfig: SubschemaConfig): Endpoint {
```

Comment:
@yaacovCR – does this look right for a shared endpoint?

Reply:
Looks close.
You need to collect all the endpoint type fields, including batchOptions and subscriber, because existence of endpoint will cause it all to read from endpoint

Reply:
https://github.com/ardatan/graphql-tools/blob/fe85dcb8376cbca06591ca0fcc6b70748c33c32a/packages/delegate/src/types.ts#L128-L140

And rootValue...